mrp_unbuild_restore_original_quant/models/mrp_unbuild.py

```diff
@@ -0,0 +1,59 @@
+# Copyright 2024 Quartile Limited
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import models
+
+
+class MrpUnbuild(models.Model):
+    _inherit = "mrp.unbuild"
+
+    def _prepare_move_line_vals(self, move, origin_move_line, taken_quantity):
+        vals = super()._prepare_move_line_vals(move, origin_move_line, taken_quantity)
+        if origin_move_line.owner_id:
+            vals["owner_id"] = origin_move_line.owner_id.id
+        return vals
+
+    def action_unbuild(self):
+        self.ensure_one()
+        if self.mo_id:
+            self = self.with_context(exact_unbuild=True)
```

Comment:
@yostashiro  I think this will skip `action_confirm()` for consumed_moves.
https://github.com/odoo/odoo/blob/d8ec7735aa8759fa94fc0223945ac9b19e05312e/addons/mrp/models/mrp_unbuild.py#L167-L168